### PR TITLE
feat: add notification handler

### PR DESCRIPTION
**Pull request by supereagle**
**What this PR does / why we need it**:

Add notification handler to receive notifications from workflow engine

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @your-reviewer

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/server/handler/v1alpha1/notification.go [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+			}
+		}
+
+		if found {
+			break
+		}
+	}
+
+	if found {
+		secretRef := pod.NewSecretRefValue()
```
Comment (zhujian7) [inline pkg/server/handler/v1alpha1/notification.go]: I'd prefer to move the ` NewSecretRefValue() ` functions to a common or util folder.

File: release/cyclone-server.yaml [hunk 2]
```diff
@@ -60,6 +60,7 @@ _config:
             "cyclone_server_host": "0.0.0.0",
             "cyclone_server_port": 7099,
             "webhook_url": "[[ compassWebEndpoint ]]/hodor/apis/admin.devops.caicloud.io/v2",
+            "record_web_url_template": "[[ compassWebEndpoint ]]/devops/pipeline/{{.WorkflowName}}/record/{{.WorkflowRunName}}?workspace={{.ProjectName}}&tenant={{.Tenant}}",
```
Comment (zhujian7) [inline release/cyclone-server.yaml]: Is the [cyclone.yaml.template](https://github.com/caicloud/cyclone/blob/master/manifests/cyclone.yaml.template) need the config ?

File: release/workflow-controller.yaml [hunk 3]
```diff
@@ -62,10 +62,5 @@ _config:
             },
             "pvc": "cyclone-server-server-v1-0-cyclone-data",
             "cyclone_server_addr": "cyclone-server.default:7099",
-            "notifications": [
-              {
-                "name": "devops",
-                "url": "http://devops-admin.default.svc.cluster.local:7088/api/v2/pipeline/notifications"
-              }
-            ]
+            "notification_url": "http://cyclone-server.default:7099/apis/v1alpha1/notifications/workflowruns"
```
Comment (zhujian7) [inline release/workflow-controller.yaml]: Is the cyclone.yaml.template need the config ?

File: pkg/server/handler/v1alpha1/notification.go (hunk 1 above)
Comment (supereagle) [reply pkg/server/handler/v1alpha1/notification.go]: Agree

Comment (supereagle) [reply pkg/server/handler/v1alpha1/notification.go]: Found there are many dependency of this function, will do this later.

**Pull request merged**